source/simulator/freesurface.cc

```diff
@@ -377,7 +379,39 @@ namespace aspect
     Vector<double> cell_vector (dofs_per_cell);
     FullMatrix<double> cell_matrix (dofs_per_cell, dofs_per_cell);
 
-    mesh_matrix = 0;
+    mesh_matrix.clear ();
+
+    //The degrees of freedom for different dimensions do not couple,
+    //so we are just solving a Laplacian for each spatial direction
```

Comment:
I think the reasoning is the other way around: "We are just solving a Laplacian for each spatial direction, so the DoFs of different components do not couple".